WordPress/gutenberg
# Gallery: Convert Gallery block to use  Image blocks instead of having its own nested image format

Fixes #10057

## Description

Refactors the Gallery block to be a wrapper around Image blocks using InnerBlocks. 

![gallery](https://user-images.githubusercontent.com/3629020/104543751-0a689100-568b-11eb-8735-dbcf2e538785.gif)

Taking this approach eliminates the current differences between stand alone image blocks, and images within the gallery, and can hopefully avoid a lot of duplication of image functionality between the various gallery blocks.

See previous [PoC PR](https://github.com/WordPress/gutenberg/pull/24039) for some background discussion 

## Testing

- This PR can be tested with adding new Galleries and making sure existing galleries still display in the old formate
- Check out the PR and add or view a gallery. It should still be in the old format
- Turn on the gutenberg experimental feature 'Gallery refactor' and any new galleries added should now be made up of nested Image blocks.
- Related PR #28481 has the block fixture and unit test updates - separated out to help make review easier

## Types of changes
Significant changes to structure of Gallery block to remove image metadata and replace with Image blocks via InnerBlocks

## Checklist:
